airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/AirbyteConnectorRunner.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk
+
+import io.airbyte.cdk.command.ConnectorCommandLinePropertySource
+import io.airbyte.cdk.integrations.base.AirbyteTraceMessageUtility
+import io.airbyte.cdk.integrations.base.JavaBaseConstants
+import io.airbyte.cdk.integrations.util.ApmTraceUtils
+import io.airbyte.cdk.integrations.util.ConnectorExceptionUtil
+import io.micronaut.configuration.picocli.MicronautFactory
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.env.CommandLinePropertySource
+import io.micronaut.context.env.Environment
+import io.micronaut.core.cli.CommandLine as MicronautCommandLine
+import java.nio.file.Path
+import picocli.CommandLine
+import picocli.CommandLine.Model.ArgGroupSpec
+import picocli.CommandLine.Model.OptionSpec
+import picocli.CommandLine.Model.UsageMessageSpec
+
+/** Source connector entry point. */
+class AirbyteSourceRunner(args: Array<out String>) :
+    AirbyteConnectorRunner(ConnectorType.SOURCE, args) {
```

Comment:
not sure I understand the need for inheritance here. That's one of the things that I would have removed from the current CDK if we were to build upon it

Reply:
Isn't inheritance OK when it's from a sealed class? It's just syntactic sugar anyways, the alternative is having both a `runSource` and a `runDestination` static fn.

Reply:
I got rid of the enum below, so sealed classes it is.